components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,540 @@
+			},
+			noPointer: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-pointer'
+			},
+			align: {
+				type: String,
+				reflect: true
+			},
+			boundary: {
+				type: Object,
+			},
+			opened: {
+				type: Boolean,
+				reflect: true
+			},
+			/**
+			 * Private.
+			 */
+			openedAbove: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'opened-above'
+			},
+			/**
+			 * Whether to render the content immediately. By default, the content rendering
+			 * is deferred.
+			 */
+			renderContent: {
+				type: Boolean,
+				attribute: 'render-content'
+			},
+			verticalOffset: {
+				type: String,
+				attribute: 'vertical-offset'
+			},
+			_width: {
+				type: Number
+			},
+			_maxHeight: {
+				type: Number
+			},
+			_position: {
+				type: Number
+			},
+			_bottomOverflow: {
+				type: Boolean
+			},
+			_topOverflow: {
+				type: Boolean
+			},
+			_contentOverflow: {
+				type: Boolean
+			}
+		};
+	}
+
+	constructor() {
+		super();
+
+		this.__opened = false;
+		this.__content = null;
+		this.__previousFocusableAncestor = null;
+		this.__applyFocus = true;
+		this.__dismissibleId = null;
+
+		this._bottomOverflow = false;
+		this._topOverflow = false;
+		this._contentOverflow = false;
+
+		this.__onResize = this.__onResize.bind(this);
+		this.__onAutoCloseFocus = this.__onAutoCloseFocus.bind(this);
+		this.__onAutoCloseClick = this.__onAutoCloseClick.bind(this);
+		this.__toggleScrollStyles = this.__toggleScrollStyles.bind(this);
+	}
+
+	get opened() {
+		return this.__opened;
+	}
+
+	set opened(val) {
+		const oldVal = this.__opened;
+		this.__opened = val;
+		this.requestUpdate('opened', oldVal);
+		this.__openedChanged(val);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.__content = this.__getContentContainer();
+		this.addEventListener('d2l-dropdown-close', this.__onClose);
+		this.addEventListener('d2l-dropdown-position', this.__toggleScrollStyles);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		window.addEventListener('resize', this.__onResize);
+		document.body.addEventListener('focus', this.__onAutoCloseFocus, true);
+		document.body.addEventListener('click', this.__onAutoCloseClick, true);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this.__onResize);
+		document.body.removeEventListener('focus', this.__onAutoCloseFocus, true);
+		document.body.removeEventListener('click', this.__onAutoCloseClick, true);
+		clearDismissible(this.__dismissibleId);
+		this.__dismissibleId = null;
+	}
+
+	updated(changedProperties) {
+		changedProperties.forEach((_, propName) => {
+			if (propName === 'verticalOffset') {
+				// for IE11
+				if (window.ShadyCSS) window.ShadyCSS.styleSubtree(this, { '--d2l-dropdown-verticaloffset': `${this.verticalOffset}px` });
+				else this.style.setProperty('--d2l-dropdown-verticaloffset', `${this.verticalOffset}px`);
+			}
+		});
+	}
+
+	_renderContent() {

```

Comment:
Nit: looks like some method are out of order in here. I am surprised that the [linting rules](https://github.com/Brightspace/eslint-config-brightspace/blob/master/sort-member-config.js) aren't causing failures for it.  I wonder if they are actually still working?

Reply:
I did get a linter error for constructor placement that it auto-fixed but other than that I haven't seem any other errors related to method order.